tools/src/main/java/it/eng/tools/configuration/ApplicationPropertiesConfiguration.java

```diff
@@ -0,0 +1,91 @@
+package it.eng.tools.configuration;
+
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.env.OriginTrackedMapPropertySource;
+import org.springframework.core.env.AbstractEnvironment;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.PropertySource;
+import org.springframework.stereotype.Component;
+
+import it.eng.tools.model.ApplicationProperty;
+import it.eng.tools.repository.ApplicationPropertiesRepository;
+import it.eng.tools.service.ApplicationPropertiesService;
+import jakarta.annotation.PostConstruct;
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class ApplicationPropertiesConfiguration {
+
+	@Autowired
+	private Environment environment;

```

Comment:
Can this be also wired through the constructor?

Reply:
True. what are the benefits?